pkg/controllers/vdb/localdatacheck_reconcile.go

```diff
@@ -0,0 +1,62 @@
+/*
+ (c) Copyright [2021-2022] Micro Focus or one of its affiliates.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ You may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package vdb
+
+import (
+	"context"
+	"fmt"
+
+	vapi "github.com/vertica/vertica-kubernetes/api/v1beta1"
+	"github.com/vertica/vertica-kubernetes/pkg/controllers"
+	"github.com/vertica/vertica-kubernetes/pkg/events"
+	corev1 "k8s.io/api/core/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+// LocalDataCheckReconciler will check the free space available in the PV and
+// log events if they it is too low.
+type LocalDataCheckReconciler struct {
+	VRec      *VerticaDBReconciler
+	Vdb       *vapi.VerticaDB
+	PFacts    *PodFacts
+	NumEvents int // Number of events written
```

Comment:
What do you use this for?

Reply:
This is for the unit test to prove that we recorded the correct number of events.